features/kernel/src/main/feature/feature.xml

```diff
@@ -23,8 +23,33 @@
                  The karaf pax.web features (version 7.0.0) are blacklisted-->
     <repository>mvn:org.ops4j.pax.web/pax-web-features/${pax.web.version}/xml/features</repository>
 
+    <!-- overwrite karaf's spring with an upgraded version to address vulnerabilities -->
```

Comment:
❓ Is there any reason we added this to the kernel feature file instead of the karaf feature file that contains other karaf feature overrides?

Reply:
This is just where I've been told to override karaf features before I'll move it to the karaf feature file if I can, since that seems preferable.

Reply:
Are those feature files that reference spring our own? If so, they can be pointed to the our karaf feature repo after moving the feature overrides there.